http-clients/src/main/java/com/palantir/remoting/http/errors/SerializableErrorErrorDecoder.java

```diff
@@ -59,12 +60,27 @@ public Exception decode(String methodKey, Response response) {
                 try {
                     SerializableError error = MAPPER.readValue(bodyAsString, SerializableError.class);
                     Class<? extends Exception> exceptionClass = error.getExceptionClass();
-                    Exception exception = exceptionClass.getConstructor(String.class).newInstance(error.getMessage());
+
+                    // Construct remote exception
+                    Exception remoteException =
+                            exceptionClass.getConstructor(String.class).newInstance(error.getMessage());
                     List<StackTraceElement> stackTrace = error.getStackTrace();
                     if (stackTrace != null) {
-                        exception.setStackTrace(stackTrace.toArray(new StackTraceElement[stackTrace.size()]));
+                        remoteException.setStackTrace(stackTrace.toArray(new StackTraceElement[stackTrace.size()]));
+                    }
+
+                    // Construct local exception that wraps the remote exception
+                    Constructor<? extends Exception> constructor;
+                    try {
+                        constructor = exceptionClass.getConstructor(String.class, Throwable.class);
+                    } catch (NoSuchMethodException e) {
+                        constructor = exceptionClass.getConstructor(String.class);
                     }
-                    return exception;
+                    Exception localException = constructor.newInstance(error.getMessage(), remoteException);
```

Comment:
doesn't this break on exception classes that only have String and not (String, Throwable) ?